components/webext-storage/sql/create_schema.sql

```diff
@@ -0,0 +1,39 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+-- This is a very simple schema.
+CREATE TABLE IF NOT EXISTS moz_extension_data (
+    ext_id TEXT PRIMARY KEY,
+    /* The JSON payload. not null because we prefer to delete the row (possibly
+       creating a tombstone) than null it */
+    data TEXT NOT NULL,
+
+    /* Same "sync status" strategy used by other components. */
+    syncStatus INTEGER NOT NULL DEFAULT 0,
+    syncChangeCounter INTEGER NOT NULL DEFAULT 1
+) WITHOUT ROWID;
+
+CREATE TABLE IF NOT EXISTS moz_extension_data_tombstones (
```

Comment:
Just out of curiosity (and so I get a better understanding of how extensions differ from other components), why have this table and not just an `is_deleted` column in `moz_extension_data`?

Reply:
> Just out of curiosity (and so I get a better understanding of how extensions differ from other components), why have this table and not just an `is_deleted` column in `moz_extension_data`?

That's a great question without a great answer :) Note also that bookmarks does, roughly, this too.

On reason is that not all deletions leave a tombstone - eg, items with SyncStatus::New were never uploaded to the server, so a deletion can actually delete. And deletions don't require any of the extra metadata in that table.

That said though, I've gone back and forward on this exact question many times - also on "data == null vs is_deleted column" particularly in the staging and mirror tables. If you have strong opinions, let me know, as mine are not!